module/VuFindSearch/src/VuFindSearch/Backend/Blender/Response/Json/RecordCollection.php

```diff
@@ -0,0 +1,441 @@
+    {
+        return $this->errors;
+    }
+
+    /**
+     * Set the source backend identifier.
+     *
+     * @param string $identifier Backend identifier
+     *
+     * @return void
+     */
+    public function setSourceIdentifier($identifier)
+    {
+        $this->source = $identifier;
+    }
+
+    /**
+     * Calculate the backend to be used for the record at a given position
+     *
+     * Note: This does not take into account whether there are enough records in the
+     * source.
+     *
+     * @param int   $position       Position
+     * @param int   $blockSize      Record block size
+     * @param array $backendIds     Available backends
+     * @param array $initialResults List of backends for initial result boosts
+     *
+     * @return string
+     */
+    protected function getBackendAtPosition(
+        int $position,
+        int $blockSize,
+        array $backendIds,
+        array $initialResults
+    ): string {
+        if ($boostBackend = $initialResults[$position] ?? false) {
+            return $boostBackend;
+        }
+
+        // We're outside the blocks affected by boosting, calculate by block
+        $currentBlock = floor($position / $blockSize);
+        $backendCount = count($backendIds);
+        return $backendCount ? $backendIds[$currentBlock % $backendCount] : '';
+    }
+
+    /**
+     * Merge facets
+     *
+     * @param array $collections Result collections
+     *
+     * @return array
+     */
+    protected function getMergedFacets(array $collections): array
+    {
+        $mergedFacets = [];
+
+        // Iterate through mappings and merge values. It is important to do it this
+        // way since multiple facets may to a single one.
+        foreach ($this->mappings['Facets']['Fields'] ?? []
+            as $facetField => $settings
+        ) {
+            $list = [];
+            foreach ($collections as $backendId => $collection) {
+                $facets = $this->convertFacets($collection);
+
+                $facetType = $settings['Type'] ?? 'normal';
+                $mappings = $settings['Mappings'][$backendId] ?? [];
+                $backendFacetField = $mappings['Field'] ?? '';
+                if (!$mappings || !$backendFacetField) {
+                    continue;
+                }
+                $valueMap = $mappings['Values'] ?? [];
+                $unmappedRule = $mappings['Unmapped'] ?? 'keep';
+                $hierarchical = $mappings['Hierarchical'] ?? false;
+                foreach ($facets[$backendFacetField] ?? [] as $value => $count) {
+                    $value = $this->convertFacetValue(
+                        $value,
+                        $facetType,
+                        $unmappedRule,
+                        $valueMap,
+                        $hierarchical
+                    );
+                    if ('' === $value) {
+                        continue;
+                    }
+
+                    if (isset($list[$value])) {
```

Comment:
Any reason we use intval when initializing but not when incrementing? Would it be worth converting this into a one-liner?

`$list[$value] = ($list[$value] ?? 0) + intval($count);`

Reply:
Good idea, fixed.